### PR TITLE
ESSIFI-96 Setup API project

### PR DESCRIPTION
**Pull request by maikel-maas**
- Basic project with Express and TypeScript setup;

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "name": "state-machine-rest-api",
+  "version": "1.0.0",
+  "description": "State Machine REST API",
+  "main": "index.js",
+  "scripts": {
+    "start": "nodemon src/index.ts",
+    "build": "tsc --project ./",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Maikel Maas",
```
Comment (sksadjad) [inline package.json:11]: change the author to Sphereon

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import express, { NextFunction, Request, Response } from 'express';
+import compression from 'compression';
+import { TEST_CONTROLLER } from './controllers/test-controller';
+import { Server } from 'http';
+
+const APP: express.Application = express();
```
Comment (sksadjad) [inline src/index.ts:6]: I'm don't really like this uppercase style. what do you guys thinks? @hrehman-sphereon and @nklomp

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
```
Comment (sksadjad) [inline tsconfig.json:5]: why so many comments?

File: package.json (hunk 1 above)
Comment (maikel-maas) [reply package.json:11]: Fixed!

File: src/index.ts (hunk 2 above)
Comment (maikel-maas) [reply src/index.ts:6]: Actually I think I went too far with the upper case style for constants. Upper case is only for **global constant values and enum values**. So, I think it's reasonable to use them only for constants that are exported. Reference: https://google.github.io/styleguide/tsguide.html#identifiers.

File: tsconfig.json (hunk 3 above)
Comment (maikel-maas) [reply tsconfig.json:5]: This is autogenerated, I have to clean it up, but I decided to keep it until I better understand what are the options that we really need.

Comment (sksadjad) [reply tsconfig.json:5]: you can use the one that we already have in pe-js

**Comment by maikel-maas** [comment]
Everything is fixed in branch feature/ESSIFI-98

**Pull request closed without merging**